resteasy/src/test/scala/com/cerner/beadledom/resteasy/InjectorProcessorSpec.scala

```diff
@@ -0,0 +1,83 @@
+package com.cerner.beadledom.resteasy
+
+import org.jboss.resteasy.spi.{Registry, ResteasyProviderFactory}
+import org.scalatest.{FunSpec, MustMatchers}
+import org.scalatest.mockito.MockitoSugar
+import com.google.inject.{AbstractModule, Guice}
+
+class InjectorProcessorSpec extends FunSpec with MockitoSugar with MustMatchers{
+
+  val registry: Registry = mock[Registry]
+  val providerFactory: ResteasyProviderFactory = mock[ResteasyProviderFactory]
+
+  describe("InjectorProcessor") {
+
+    describe("constructor") {
+      it("throws a NullPointerException for null registry") {
+        intercept[NullPointerException] {
+          new InjectorProcessor(null, providerFactory)
+        }
+      }
+
+      it("throws a NullPointerException for null provider factory") {
+        intercept[NullPointerException] {
+          new InjectorProcessor(registry, null)
+        }
+      }
+    }
+
+    describe("processInjector") {
+      it("throws a NullPointerException for null injector") {
+        intercept[NullPointerException] {
+          new InjectorProcessor(registry, providerFactory).process(null)
+        }
+      }
+
+      it("excludes providers that have a binding annotation") {
+        val injector = Guice.createInjector(new AbstractModule {
+          override def configure(): Unit = {
+            bind(classOf[TestJaxRsProvider]).annotatedWith(classOf[TestBindingAnnotation]).to(
+              classOf[TestJaxRsProvider]
+            )
+          }
+        })
+        val providerFactory = new ResteasyProviderFactory()
+        new InjectorProcessor(registry, providerFactory).process(injector)
+
+        providerFactory.getProviderInstances mustBe empty
+      }
+
+      it("excludes provider with binding annotation when two providers of the same type are bound with the injector") {
+        val injector = Guice.createInjector(new AbstractModule {
+          override def configure(): Unit = {
+            bind(classOf[TestJaxRsProvider]).annotatedWith(classOf[TestBindingAnnotation]).to(
+              classOf[TestJaxRsProvider]
+            )
+
+            bind(classOf[TestJaxRsProvider]).toProvider(new TestProvider())
```

Comment:
I think this line could be just `bind(classOf[TestJaxRsProvider])` and would still work. Then the `TestProvider` class wouldn't need to exist.

Reply:
https://github.com/cerner/beadledom/pull/143/commits/bd6ab90bbf3a2dfb902bc7cda220fbd0a1cae674